src/components/TreeView/TreeNode.tsx

```diff
@@ -0,0 +1,28 @@
+import * as React from 'react';
+import { themr, ThemedComponentClass } from 'react-css-themr';
+// import { classNames } from '@shopify/react-utilities/styles';
+
+import { SourceData as Props } from './interface';
+import { TREEVIEW } from '../ThemeIdentifiers';
+
+import * as baseTheme from './TreeView.scss';
+
+class TreeNode extends React.Component<Props, never> {
+  constructor(props: Props) {
+    super(props);
+  }
+
+  renderNode = (): any => {
+    const { label = '' } = this.props;
+
+    return label;
+  }
+
+  render() {
+    // If component prop is not present then check for label & render that
+    const { component = this.renderNode } = this.props;
```

Comment:
@dheeraja00 Can you help me understand the syntax here? `{ component = this.renderNode }` if component is null, then it sets it to renderNode, else it doesn't.  Is this correct?